dev_tools/nyaml2nxdl/nyaml2nxdl_backward_tools.py

```diff
@@ -0,0 +1,1046 @@
+            # more indentation.
+            file_out.write(
+                f"{depth_ * DEPTH_SIZE}{key}: "
+                f"{handle_mapping_char(val, depth_ + 1, False)}\n"
+            )
+
+    def handel_link(self, depth, node, file_out):
+        """
+        Handle link elements of nxdl
+        """
+
+        possible_link_attrs = ["name", "target", "napimount"]
+        node_attr = node.attrib
+        # Handle special cases
+        if "name" in node_attr:
+            file_out.write(
+                "{indent}{name}(link):\n".format(
+                    indent=depth * DEPTH_SIZE, name=node_attr["name"] or ""
+                )
+            )
+            del node_attr["name"]
+
+        depth_ = depth + 1
+        # Handle general cases
+        for attr_key, val in node_attr.items():
+            if attr_key in possible_link_attrs:
+                file_out.write(
+                    "{indent}{attr}: {value}\n".format(
+                        indent=depth_ * DEPTH_SIZE, attr=attr_key, value=val
+                    )
+                )
+            else:
+                raise ValueError(
+                    f"An anexpected attribute '{attr_key}' of link has found."
+                    f"At this moment the alloed keys are {possible_link_attrs}"
+                )
+
+    def handel_choice(self, depth, node, file_out):
+        """
+        Handle choice element which is a parent node of group.
+        """
+
+        possible_attr = []
+
+        node_attr = node.attrib
+        # Handle special casees
+        if "name" in node_attr:
+            file_out.write(
+                "{indent}{attr}(choice): \n".format(
+                    indent=depth * DEPTH_SIZE, attr=node_attr["name"]
+                )
+            )
+            del node_attr["name"]
+
+        depth_ = depth + 1
+        # Taking care of general attrinutes. Though, still no attrinutes have found,
+        # but could be used for future
+        for attr in node_attr.items():
+            if attr in possible_attr:
+                file_out.write(
+                    "{indent}{attr}: {value}\n".format(
+                        indent=depth_ * DEPTH_SIZE, attr=attr, value=node_attr[attr]
+                    )
+                )
+            else:
+                raise ValueError(
+                    f"An unexpected attribute '{attr}' of 'choice' has been found."
+                    f"At this moment attributes for choice {possible_attr}"
+                )
+
+    def handel_comment(self, depth, node, file_out):
+        """
+        Collect comment element and pass to write_out function
+        """
+        indent = depth * DEPTH_SIZE
+        if self.is_last_element_comment:
+            text = self.comvert_to_ymal_comment(indent, node.text)
+            self.write_out(indent, text, file_out)
+        else:
+            text = self.comvert_to_ymal_comment(indent, node.text)
+            self.write_out(indent, text, file_out)
```

Comment:
Common lines so move before `if` clause

Reply:
Not a common line, `depth_` already been used in preceded `if clase`.